### ISSUE TITLE
Allow a way to select the behaviour of 0 in the sign() function. 

### ISSUE THREAD

**Issue by yosoyfreeman** [godotengine/godot-proposals#6854]
### Describe the project you are working on

A triplanar shader.

### Describe the problem or limitation you are having in your project

Right now using the sign function (Shader or visual shaders) will flicker in almost any use case.

Sign function is problematic by itself when passed values of 0.  Is designed to return -1 or 1,  so 3D applications must decide what happens when the value is neither positive or negative. Right now, Godot uses the less useful solution, which is to return  0.  This makes the function virtually unusable for most of the use cases someone may need it.

More info in the bug report i made: https://github.com/godotengine/godot/issues/76909

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

This change would make the sign function more usable in general terms and allow Godot users learning shaders to follow tutorials of any other 3D engine. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

-Change the default behaviour to return either -1 or 1 as one wold expect.

-Even better, allow to select the behavior or values who are 0: Return 0, -1 or 1.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

If possible, is difficult to do.

### Is there a reason why this should be core and not an add-on in the asset library?

This is a usability problem. Should be core.

**Comment by AThousandShips** [comment]
To be clear that this is *not* what you would expect if you are a mathematician, or use the function in GDScript with the same name, so the assumption that everyone assumes this is how `sign` works is not really true

It is true that `sign` is defined somewhat confusingly across different programming languages and contexts, but the mathematical definition is clear and standardized

Adding functionality to allow for margins would be useful though

**Comment by yosoyfreeman** [comment]
> To be clear that this is _not_ what you would expect if you are a mathematician, or use the function in GDScript with the same name, so the assumption that everyone assumes this is how `sign` works is not really true

You are totally right. I was referring specifically to it use on shaders. Is usually not the way is treated.

> Why not keep `sign(val)` as is and add a new function like `abs_sign(val, 1)` which takes the value and the value to apply if zero?

I proposed adding it to the function to reduce cluster, specially in visual nodes but that would work perfectly.

**Comment by adamscott** [comment]
> -Even better, allow to select the behavior or values who are 0: Return 0, -1 or 1.

Didn't read that line @yosoyfreeman. Sorry!

**Comment by yosoyfreeman** [comment]
> > -Even better, allow to select the behavior or values who are 0: Return 0, -1 or 1.
> 
> Didn't read that line @yosoyfreeman. Sorry!

Don't worry. Maybe the function could have a parameter who is 0 by default, so using sign() alone does exactly what mathematically should do, and something like sign(Value to add to 0) to override it.

**Comment by RedworkDE** [comment]
> In unity or unreal values that are 0 are either included as positive or as negative. Same for any other software.

Please give an actual example where sign(0) is not 0.

Both Unity and Unreal use HLSL for their shaders where `sign(0.0) == 0.0`: https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-sign
Same for GLSL, where `sign` also returns 0 for 0 inputs: https://registry.khronos.org/OpenGL-Refpages/gl4/html/sign.xhtml

**Comment by yosoyfreeman** [comment]
> > In unity or unreal values that are 0 are either included as positive or as negative. Same for any other software.
> 
> Please give an actual example where sign(0) is not 0.
> 
> Both Unity and Unreal use HLSL for their shaders where `sign(0.0) == 0.0`: https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-sign Same for GLSL, where `sign` also returns 0 for 0 inputs: https://registry.khronos.org/OpenGL-Refpages/gl4/html/sign.xhtml

I'm not afraid to say i was wrong on that. I was trying to understand the issue. But the addition would be still useful i think.

Now, the flickering issue i showed here and the bug issue is in fact a bug. As you said, it should give 0, but it fluctuates between -1,0 and 1. I think is related to numerical precision and what is considered 0 should have a margin to account for it so it correctly reports 0 instead of flickering between the 3 states.

**Comment by yosoyfreeman** [comment]
Is there already: https://github.com/godotengine/godot/issues/76909
